diff -up bind-9.5.0/lib/isccfg/aclconf.c.450995 bind-9.5.0/lib/isccfg/aclconf.c
--- bind-9.5.0/lib/isccfg/aclconf.c.450995	2008-06-18 12:29:05.000000000 +0200
+++ bind-9.5.0/lib/isccfg/aclconf.c	2008-06-18 12:30:01.000000000 +0200
@@ -287,6 +287,7 @@ nested_acl:
 			} else {
 				dns_acl_merge(dacl, inneracl,
 					      ISC_TF(!neg));
+				de += inneracl->length;  /* elements added */
 				dns_acl_detach(&inneracl);
 				continue;
 			}
