diff -up bind-9.5.1b1/bin/named/include/named/globals.h.varrun bind-9.5.1b1/bin/named/include/named/globals.h
--- bind-9.5.1b1/bin/named/include/named/globals.h.varrun	2007-09-26 05:22:43.000000000 +0200
+++ bind-9.5.1b1/bin/named/include/named/globals.h	2008-07-21 12:13:33.000000000 +0200
@@ -107,9 +107,9 @@ EXTERN isc_boolean_t		ns_g_foreground		I
 EXTERN isc_boolean_t		ns_g_logstderr		INIT(ISC_FALSE);
 
 EXTERN const char *		ns_g_defaultpidfile 	INIT(NS_LOCALSTATEDIR
-							     "/run/named.pid");
+							     "/run/named/named.pid");
 EXTERN const char *		lwresd_g_defaultpidfile INIT(NS_LOCALSTATEDIR
-							    "/run/lwresd.pid");
+							    "/run/named/lwresd.pid");
 EXTERN const char *		ns_g_username		INIT(NULL);
 
 EXTERN int			ns_g_listen		INIT(3);
