diff -up bind-9.7.1b1/lib/isccfg/namedconf.c.keysdir bind-9.7.1b1/lib/isccfg/namedconf.c
--- bind-9.7.1b1/lib/isccfg/namedconf.c.keysdir	2010-05-21 14:53:47.547846620 +0200
+++ bind-9.7.1b1/lib/isccfg/namedconf.c	2010-05-21 14:55:37.587846690 +0200
@@ -849,6 +849,7 @@ options_clauses[] = {
 	{ "interface-interval", &cfg_type_uint32, 0 },
 	{ "listen-on", &cfg_type_listenon, CFG_CLAUSEFLAG_MULTI },
 	{ "listen-on-v6", &cfg_type_listenon, CFG_CLAUSEFLAG_MULTI },
+	{ "managed-keys-directory", &cfg_type_qstring, 0 },
 	{ "match-mapped-addresses", &cfg_type_boolean, 0 },
 	{ "memstatistics-file", &cfg_type_qstring, 0 },
 	{ "memstatistics", &cfg_type_boolean, 0 },
