diff -up bind-9.5.0b1/lib/dns/rbtdb.c.segv bind-9.5.0b1/lib/dns/rbtdb.c
--- bind-9.5.0b1/lib/dns/rbtdb.c.segv	2008-02-11 14:52:12.000000000 +0100
+++ bind-9.5.0b1/lib/dns/rbtdb.c	2008-02-11 14:55:58.000000000 +0100
@@ -1633,7 +1633,8 @@ decrement_reference(dns_rbtdb_t *rbtdb, 
                                       "decrement_reference: "
                                       "dns_rbt_deletenode: %s",
                                       isc_result_totext(result));
-        } else if (dns_rbtnode_refcurrent(node) == 0) {
+        } else if (rbtdb->deadnodes != NULL &&
+		   dns_rbtnode_refcurrent(node) == 0) {
                 INSIST(!ISC_LINK_LINKED(node, deadlink));
                 ISC_LIST_APPEND(rbtdb->deadnodes[bucket], node, deadlink);
         }
