diff -up bind-9.6.0b1/configure.in.libxml2 bind-9.6.0b1/configure.in
--- bind-9.6.0b1/configure.in.libxml2	2008-11-11 16:04:27.000000000 +0100
+++ bind-9.6.0b1/configure.in	2008-11-11 16:04:46.000000000 +0100
@@ -1002,7 +1002,7 @@ case "$use_libxml2" in
 		;;
 	auto|yes)
 		case X`(xml2-config --version) 2>/dev/null` in
-		X2.6.*)
+		X2.[[67]].*)
 			libxml2_libs=`xml2-config --libs`
 			libxml2_cflags=`xml2-config --cflags`
 			;;
