Index: make/rules.in
===================================================================
RCS file: /proj/cvs/prod/bind9/make/rules.in,v
retrieving revision 1.62
diff -u -r1.62 rules.in
--- make/rules.in       18 Jun 2007 23:47:57 -0000      1.62
+++ make/rules.in       18 Feb 2008 14:46:16 -0000
@@ -104,7 +104,7 @@
 BUILD_CC = @BUILD_CC@
 BUILD_CFLAGS = @BUILD_CFLAGS@
 BUILD_CPPFLAGS = @BUILD_CPPFLAGS@
-BUILD_LDFAGS = @BUILD_LDFAGS@
+BUILD_LDFLAGS = @BUILD_LDFLAGS@
 BUILD_LIBS = @BUILD_LIBS@

 .SUFFIXES:
