--- bind-9.4.0/bin/named/Makefile.in.dbus_archdep_libdir	2007-03-06 14:18:14.000000000 +0100
+++ bind-9.4.0/bin/named/Makefile.in	2007-03-06 14:20:39.000000000 +0100
@@ -43,8 +43,9 @@
 		${ISCCFG_INCLUDES} ${ISCCC_INCLUDES} ${ISC_INCLUDES} \
 		${DLZDRIVER_INCLUDES} ${DBDRIVER_INCLUDES}
 
+DBUS_ARCHDEP_LIBDIR ?= lib
 DBUS_INCLUDES = \
-        -I/usr/lib/dbus-1.0/include -I/usr/include/dbus-1.0
+        -I/usr/${DBUS_ARCHDEP_LIBDIR}/dbus-1.0/include -I/usr/include/dbus-1.0
 
 CDEFINES =      @USE_DLZ@
 
