--- bind-9.3.1beta2/configure.in.openssl_suffix	2004-12-08 23:07:10.000000000 -0500
+++ bind-9.3.1beta2/configure.in	2005-01-27 17:03:49.394814000 -0500
@@ -374,6 +374,10 @@
 		fi
 	done
 fi
+OSSUFFIX=
+if test "$host_cpu" = "x86_64"; then
+        OSSUFFIX=64
+fi;
 case "$use_openssl" in
 	no)
 		AC_MSG_RESULT(no)
@@ -416,11 +420,11 @@
 				DNS_OPENSSL_LIBS="-L$use_openssl/lib -R$use_openssl/lib -lcrypto"
 				;;
 			*)
-				DNS_OPENSSL_LIBS="-L$use_openssl/lib -lcrypto"
+				DNS_OPENSSL_LIBS="-L$use_openssl/lib${OSUFFIX} -lcrypto"
 				;;
 			esac
 		fi
-		AC_MSG_RESULT(using openssl from $use_openssl/lib and $use_openssl/include)
+		AC_MSG_RESULT(using openssl from $use_openssl/lib${OSSUFFIX} and $use_openssl/include)
 
 		saved_cflags="$CFLAGS"
 		saved_libs="$LIBS"
