--- bind-9.5.0a5/bin/named/Makefile.in.xml	2007-06-19 11:49:35.000000000 +0200
+++ bind-9.5.0a5/bin/named/Makefile.in	2007-06-19 11:49:48.000000000 +0200
@@ -154,7 +154,7 @@ bind9.xsl.h: bind9.xsl convertxsl.pl
 	${PERL} ${srcdir}/convertxsl.pl < ${srcdir}/bind9.xsl > bind9.xsl.h
 
 depend: bind9.xsl.h
-server.@O@: bind9.xsl.h
+server.o: bind9.xsl.h
 
 installdirs:
 	$(SHELL) ${top_srcdir}/mkinstalldirs ${DESTDIR}${sbindir}
