diff -up bind-9.5.0b1/lib/dns/include/dst/Makefile.in.gssapi-header bind-9.5.0b1/lib/dns/include/dst/Makefile.in
--- bind-9.5.0b1/lib/dns/include/dst/Makefile.in.gssapi-header	2007-12-11 15:50:55.000000000 +0100
+++ bind-9.5.0b1/lib/dns/include/dst/Makefile.in	2007-12-11 15:51:15.000000000 +0100
@@ -21,7 +21,7 @@ top_srcdir =	@top_srcdir@
 
 @BIND9_VERSION@
 
-HEADERS =	dst.h lib.h result.h
+HEADERS =	dst.h gssapi.h lib.h result.h
 
 SUBDIRS =
 TARGETS =
