--- bind-9.4.0/bin/dnssec/dnssec-signzone.c.directory	2007-04-11 15:42:56.000000000 +0200
+++ bind-9.4.0/bin/dnssec/dnssec-signzone.c	2007-04-11 15:43:23.000000000 +0200
@@ -1470,8 +1470,8 @@
 		fatal("failed to find the zone's origin: %s",
 		      isc_result_totext(result));
 
-	result = dns_dnssec_findzonekeys(db, currentversion, node, gorigin,
-					 mctx, 20, keys, &nkeys);
+	result = dns_dnssec_findzonekeys2(db, currentversion, node, gorigin,
+					 directory, mctx, 20, keys, &nkeys);
 	if (result == ISC_R_NOTFOUND)
 		result = ISC_R_SUCCESS;
 	if (result != ISC_R_SUCCESS)
